activesupport/test/file_update_checker_shared_tests.rb

```diff
@@ -0,0 +1,240 @@
+require 'fileutils'
+
+module FileUpdateCheckerSharedTests
+  include FileUtils
+
+  def tmpdir
+    @tmpdir ||= Dir.mktmpdir(nil, __dir__)
+  end
+
+  def tmpfile(name)
+    "#{tmpdir}/#{name}"
+  end
+
+  def tmpfiles
+    @tmpfiles ||= %w(foo.rb bar.rb baz.rb).map {|f| tmpfile(f)}
+  end
+
+  def teardown
+    FileUtils.rm_rf(@tmpdir) if @tmpdir
+  end
+
+  def test_should_not_execute_the_block_if_no_paths_are_given
```

Comment:
Since we moved all the methods to this file what do you think about using the same style of test description in both tests. Or `def test_` or `test 'bla' do`